console_test.go

```diff
@@ -533,20 +546,37 @@ func TestConsoleWriterConfiguration(t *testing.T) {
 		}
 	})
 
-	t.Run("Sets FieldsOrder", func(t *testing.T) {
+	t.Run("Sets FieldsOrder sorted extras", func(t *testing.T) {
 		buf := &bytes.Buffer{}
 		w := zerolog.ConsoleWriter{Out: buf, NoColor: true, FieldsOrder: []string{"zebra", "aardvark"}}
 
-		evt := `{"level": "info", "message": "Zoo", "aardvark": "Able", "mussel": "Mountain", "zebra": "Zulu"}`
+		evt := `{"level": "info", "message": "Zoo", "aardvark": "Able", "mussel": "Mountain", "zebra": "Zulu", "lizard": "Larry"}`
 		_, err := w.Write([]byte(evt))
 		if err != nil {
 			t.Errorf("Unexpected error when writing output: %s", err)
 		}
 
-		expectedOutput := "<nil> INF Zoo zebra=Zulu aardvark=Able mussel=Mountain\n"
+		expectedOutput := "<nil> INF Zoo zebra=Zulu aardvark=Able lizard=Larry mussel=Mountain\n"
 		actualOutput := buf.String()
 		if actualOutput != expectedOutput {
-			t.Errorf("Unexpected output %q, want: %q", actualOutput, expectedOutput)
+			t.Errorf("Unexpected output\ngot:  %q\nwant: %q\n", actualOutput, expectedOutput)
+		}
+	})
+
+	t.Run("Sets FieldsOrder unsorted extras", func(t *testing.T) {
+		buf := &bytes.Buffer{}
+		w := zerolog.ConsoleWriter{Out: buf, NoColor: true, FieldsOrder: []string{"zebra", "aardvark"}, DoNotSortFields: true}
```

Comment:
Passing `DoNotSortFields: true` keeps fields in original order _except as far as `FieldsOrder` mentions_.